File: Packages/RallyCore/Sources/RallyCore/Models/School.swift
import Foundation

/// Represents a partner school in the Rally platform.
public struct School: Identifiable, Codable, Hashable, Sendable {
    public let id: String
    public let name: String
    public let mascot: String
    public let abbreviation: String
    public let logoURL: URL?
    public let mascotImageURL: URL?
    public let bannerImageURL: URL?
    public let theme: SchoolTheme
    public let venues: [Venue]
    public let isActive: Bool

    public init(
        id: String,
        name: String,
        mascot: String,
        abbreviation: String,
        logoURL: URL? = nil,
        mascotImageURL: URL? = nil,
        bannerImageURL: URL? = nil,
        theme: SchoolTheme,
        venues: [Venue] = [],
        isActive: Bool = true
    ) {
        self.id = id
        self.name = name
        self.mascot = mascot
        self.abbreviation = abbreviation
        self.logoURL = logoURL
        self.mascotImageURL = mascotImageURL
        self.bannerImageURL = bannerImageURL
        self.theme = theme
        self.venues = venues
        self.isActive = isActive
    }
}

public struct SchoolTheme: Codable, Hashable, Sendable {
    public let primaryColor: String
    public let secondaryColor: String
    public let accentColor: String
    public let darkModeBackground: String?
    public let fontOverride: String?

    public init(
        primaryColor: String,
        secondaryColor: String,
        accentColor: String,
        darkModeBackground: String? = nil,
        fontOverride: String? = nil
    ) {
        self.primaryColor = primaryColor
        self.secondaryColor = secondaryColor
        self.accentColor = accentColor
        self.darkModeBackground = darkModeBackground
        self.fontOverride = fontOverride
    }
}

public struct Venue: Identifiable, Codable, Hashable, Sendable {
    public let id: String
    public let name: String
    public let latitude: Double
    public let longitude: Double
    public let radiusMeters: Double
    public let beaconUUID: String?
    public let beaconMajor: UInt16?
    public let sport: Sport

    public init(
        id: String,
        name: String,
        latitude: Double,
        longitude: Double,
        radiusMeters: Double = 500,
        beaconUUID: String? = nil,
        beaconMajor: UInt16? = nil,
        sport: Sport = .football
    ) {
        self.id = id
        self.name = name
        self.latitude = latitude
        self.longitude = longitude
        self.radiusMeters = radiusMeters
        self.beaconUUID = beaconUUID
        self.beaconMajor = beaconMajor
        self.sport = sport
    }
}

public enum Sport: String, Codable, Hashable, Sendable, CaseIterable {
    case football
    case basketball
    case baseball
    case softball
    case soccer
    case volleyball
    case hockey
    case lacrosse
    case other
}

File: Android/core/src/main/java/com/rally/app/core/model/School.kt
package com.rally.app.core.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/**
 * Represents a partner school in the Rally platform.
 */
@Serializable
data class School(
    val id: String,
    val name: String,
    val mascot: String,
    val abbreviation: String,
    val logoURL: String? = null,
    val mascotImageURL: String? = null,
    val bannerImageURL: String? = null,
    val theme: SchoolTheme,
    val venues: List<Venue> = emptyList(),
    val isActive: Boolean = true
)

/**
 * Theme configuration for a school's branded experience.
 */
@Serializable
data class SchoolTheme(
    val primaryColor: String,
    val secondaryColor: String,
    val accentColor: String,
    val darkModeBackground: String? = null,
    val fontOverride: String? = null
)

/**
 * Represents a physical venue where events take place.
 */
@Serializable
data class Venue(
    val id: String,
    val name: String,
    val latitude: Double,
    val longitude: Double,
    val radiusMeters: Double = 500.0,
    val beaconUUID: String? = null,
    val beaconMajor: Int? = null,
    val sport: Sport = Sport.FOOTBALL
)

/**
 * Supported sport types in the Rally platform.
 */
@Serializable
enum class Sport {
    @SerialName("football") FOOTBALL,
    @SerialName("basketball") BASKETBALL,
    @SerialName("baseball") BASEBALL,
    @SerialName("softball") SOFTBALL,
    @SerialName("soccer") SOCCER,
    @SerialName("volleyball") VOLLEYBALL,
    @SerialName("hockey") HOCKEY,
    @SerialName("lacrosse") LACROSSE,
    @SerialName("other") OTHER;

    val displayName: String
        get() = name.lowercase().replaceFirstChar { it.uppercase() }
}
